### DISCUSSION TITLE
x-trap.noscroll doesn't preserve current scroll position

### DISCUSSION THREAD

**Discussion by jobvite-github** [alpinejs/alpine#2601] [7. Bugs (archived)]
Using `x-trap.noscroll`, it keeps taking me to the top of the page when I open the modal with x-trap.

**Comment by SimoTod** [comment]
The example in the documentation (https://alpinejs.dev/plugins/focus) doesn't behave that way. It's probably worth sharing an example to replicate the bug.

**Reply by jobvite-github** [reply]
Thanks! Working on replicating, but you're right. It doesn't seem to be an issue in the documentation, or with a simple codepen example. This may simply be an issue with how I've implemented the functionality, as it's nested within complex SVG code.

The click events work fine, and they are attached to `<g>` tags in the SVG. The `x-trap` is associated with a parent-relative element, being the modal. The functionality works, but when I add the `x-trap.noscroll`, it jumps to the top of the page. Here is a basic example of what all of this looks like:

```
<div class="map-container">
    <div id="map" x-data="{open: false}" @keydown.escape.window="open = false">
        <svg class="svgmap">
            <g id="pin-singapore" class="pin" :class="{ 'active': active == 'singapore' }" @click.stop="open = 'singapore'" @click.away="open = false" x-ref="singapore">
            <g id="pin-bangalore" class="pin" :class="{ 'active': active == 'bangalore' }" @click.stop="open = 'bangalore'" @click.away="open = false" x-ref="bangalore">
            <g id="pin-montreal" class="pin" :class="{ 'active': active == 'montreal' }" @click.stop="open = 'montreal'" @click.away="open = false" x-ref="montreal">
        </svg>
        
        <div class="modal flex-row flex-m-center flex-c-center" role="dialog" tabindex="-1" x-show.transition.opacity.500="open" x-trap.noscroll="open" x-cloak>
            <div class="modal__content">
                <button @click="open = false"><i class="far fa-times-circle"></i></button>
            </div>
        </div>
    </div>
</div>
```

Unfortunately, the current working example is included in a website not viewable publicly, yet. But, that's about the gist of the functional code.

**Reply by SimoTod** [reply]
The svg shouldn't be relevant. where/how is the modal positioned? can you share the css rules for `.modal`, `.flex-row`, `.flex-m-center`, `.flex-c-center` snd `.modal__content`. I suspect it's something to do with `position: absolute` or something like that.

**Reply by jobvite-github** [reply]
It's a fixed position. Here is a general stylesheet that reflects what I have:
```
.modal {
    position: fixed;
    width: 100%;
    height: 100%;
    z-index: $z-modal;
    top: 0;
    overflow: hidden;
}
```

**Reply by philwolstenholme** [reply]
https://css-tricks.com/prevent-page-scrolling-when-a-modal-is-open might be useful, depending on how you've set your modal up?
